### PR TITLE
[#137409] handle missing reserve end at

### PR DESCRIPTION
**Pull request by meara**
Don't try to estimate reservation cost unless both reservation start and end times exist.

### CODE REVIEW

File: spec/models/instrument_price_policy_calculations_spec.rb [hunk 1]
```diff
@@ -43,6 +43,10 @@
       expect(policy.estimate_cost_and_subsidy(end_at, start_at)).to be_nil
     end
 
+    it "returns nil if start_at or end_at is nil" do
```
Comment (jhanggi) [inline spec/models/instrument_price_policy_calculations_spec.rb]: Should we do both sides of this, i.e. nil for start, and then another test for nil for end? Also, using start_at as the end_at argument is a little confusing (the previous test does it for the sake of intentionally inverting the times).

**Comment by meara** [comment]
@jhanggi thoughts on merging this for now and making a separate task for date validation? Both seem worth doing IMO.
